plugins/woocommerce/src/StoreApi/Schemas/V1/AbstractAddressSchema.php

```diff
@@ -119,7 +119,9 @@ public function sanitize_callback( $address, $request, $param ) {
 		$sanitization_util = new SanitizationUtils();
 		$address           = (array) $address;
 		$field_schema      = $this->get_properties();
-		$address           = array_reduce(
+		// omit all keys from address that are not in the schema. This should account for email.
```

Comment:
Did we consider the risks of removing other extra fields outside of the email?
It's not obvious from the comment why the email is ok to be removed.
More details p1710347782249489/1710347365.401499-slack-C8X6Q7XQU

Reply:
AbstractAddressSchema operates on the base address (first_name,last_name,address,address_2,company,city,state,country,postcode,phone, additional fields registered to address).

Those are the fields we support in schema and therefore the fields we sanitize and validate, email is not part of this class, it's part of BillingAddressSchema.

Email is the only example we have in our codebase of a field that exists in billing and not in shipping.

Reply:
As a follow up to this. We're not breaking any backward compatibility here, because previously, we would cherry-pick fields regardless to save them ([see code here](https://github.com/woocommerce/woocommerce//blob/fcdbc76414c9e16a5d82dafc996c19d0b2f2f679/plugins/woocommerce/src/StoreApi/Routes/V1/CartUpdateCustomer.php#L160-L184)).

The code above isn't limited to email, but email is the only good example we have.